stdlib/messaging/activemq-artemis/src/main/java/org/ballerinalang/messaging/artemis/externimpl/consumer/Start.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+
+package org.ballerinalang.messaging.artemis.externimpl.consumer;
+
+import org.apache.activemq.artemis.api.core.client.ClientSession;
+import org.ballerinalang.bre.Context;
+import org.ballerinalang.bre.bvm.BlockingNativeCallableUnit;
+import org.ballerinalang.messaging.artemis.ArtemisConstants;
+import org.ballerinalang.messaging.artemis.ArtemisUtils;
+import org.ballerinalang.model.types.TypeKind;
+import org.ballerinalang.model.values.BMap;
+import org.ballerinalang.model.values.BValue;
+import org.ballerinalang.natives.annotations.BallerinaFunction;
+import org.ballerinalang.natives.annotations.Receiver;
+
+import java.util.concurrent.CountDownLatch;
+
+/**
+ * Extern function to start the Artemis consumer.
+ *
+ * @since 0.995
+ */
+
+@BallerinaFunction(
+        orgName = ArtemisConstants.BALLERINA, packageName = ArtemisConstants.ARTEMIS,
+        functionName = "start",
+        receiver = @Receiver(type = TypeKind.OBJECT, structType = ArtemisConstants.LISTENER_OBJ,
+                             structPackage = ArtemisConstants.PROTOCOL_PACKAGE_ARTEMIS)
+)
+public class Start extends BlockingNativeCallableUnit {
+    private CountDownLatch countDownLatch = new CountDownLatch(1);
+
+    @Override
+    public void execute(Context context) {
+        try {
+            @SuppressWarnings(ArtemisConstants.UNCHECKED)
+            BMap<String, BValue> listenerObj = (BMap<String, BValue>) context.getRefArgument(0);
+            listenerObj.addNativeData(ArtemisConstants.COUNTDOWN_LATCH, countDownLatch);
+            @SuppressWarnings(ArtemisConstants.UNCHECKED)
+            BMap<String, BValue> sessionObj = (BMap<String, BValue>) listenerObj.get("session");
+            ClientSession session = (ClientSession) sessionObj.getNativeData(ArtemisConstants.ARTEMIS_SESSION);
+            session.start();
+            new Thread(() -> {
+                try {
+                    countDownLatch.await();
```

Comment:
Why we can't use NativeCallableUnit and implement this in nonblocking manner?

Reply:
If we do that the start function never returns. In integration tests service deployment is checked using returning of this function. If it does not return the integration test suite hangs. Hence this is the only viable solution we could think of. @lankavitharana might be able to explain.

Reply:
Can you explain the functionality of this start method? Shall we schedule a code review for this? I think that will be much easier to discuss this. Main implementation shouldn't depend on outside factors like test suites.

Reply:
As discussed offline I have added a developer comment